csrgen.py

```diff
@@ -0,0 +1,190 @@
+                    CN = domain.strip().lower()
+                    domain = fp.readline()
+                    count += 1
+                elif 1 <= count <= 100:
+                    SANS.append(domain.strip().lower())
+                    domain = fp.readline()
+                    count += 1
+                else:
+                    print("Too many domains! Please limit to 100")
+                    exit(1)
+            print(CN)
+    else:
+        domain = args.domain
+        CN = domain.strip().lower()
+
+    generated_csr_config = open(args.domain + '_csr_config', 'w')
+    generated_csr_config.write(tm.render(cn=CN, sans=SANS))
+    generated_csr_config.close()
+
+    return str(args.domain + '_csr_config')
+
+
+def main(args):
+    if args.command == 'csr':
+        # Look for csrgen config files
+        config_file_search = fnmatch.filter(os.listdir(), "*_gen_config")
+        if len(config_file_search) == 0:
+            print("No config files found please run with 'config -h' to create one")
+            exit()
+        elif len(config_file_search) > 1:   # Check if there is more then one config file
+            if args.org:
+                # Check if any of the config files match the specified Org
+                if check_file({args.org + "_gen_config"}):
+                    csr_config_out = gen_csr_config(
+                        args, {args.org + "_gen_config"})
+            else:
+                print("Please specify Organization name with -o <Org>")
+                exit()
+        else:  # There is only one found so assume its the right one\
+            csr_config_out = gen_csr_config(args, config_file_search[0])
+
+        # Validate key argument and file
+        if args.key:
+            if check_file(args.key):
+                try:
+                    subprocess.call(['openssl', 'rsa', '-in',
+                                    args.key, '-check'])
+                except:
+                    print("Key specified is not a valid rsa key.")
+            else:
+                print("Key file specified doesn't exist.")
+        else:
+            print("No key specified. Creating a new one.")
+            subprocess.call(['openssl', 'genrsa', '-out',
+                             args.domain + '.key', '2048'])
+            print("Key Generated")
+
+        if args.key:
+            print("Generating CSR")
+            subprocess.call(['openssl', 'req', '-new', '-config', csr_config_out,
+                             '-key',  + args.key, '-out', args.domain + '.csr'])
+        else:
+            print("Generating CSR")
+            subprocess.call(['openssl', 'req', '-new', '-config', csr_config_out,
+                             '-key', args.domain + '.key', '-out', args.domain + '.csr'])
+
+    if args.command == 'config':
+        tm = Template(config_template)
+        config_file_name = args.org + "_gen_config"
+        print("Creating config template file: {filename}".format(
```

Comment:
NIT use an f-string instead of `"".format()`

ex. `print(f"creating {config_file_name}")`